ARExperiment/ARExperiment/SizeComparisonViewController.swift

```diff
@@ -39,57 +39,63 @@ class SizeComparisonViewController: UIViewController {
             return
         }
 
-        if let nodeExists = sceneView.scene.rootNode.childNode(withName: cubeName, recursively: true) {
-            nodeExists.removeFromParentNode()
+       removeNodeIfExistAlready()
+
+        guard let hitTransform = arModel.getTransformForAnchor(location: location,
+                                                         sceneView: sceneView,
+                                                         resultType: [.featurePoint]) else {
+                                                            return
         }
+        let anchor = ARAnchor(transform: hitTransform)
+        sceneView.session.add(anchor: anchor)
+    }
 
-        let hitResultsFeaturePoints: [ARHitTestResult] =
-            sceneView.hitTest(location, types: .featurePoint)
+    private func removeNodeIfExistAlready() {
+        if let nodeExists = sceneView.scene.rootNode.childNode(withName: currentCubeName, recursively: true) {
+            nodeExists.removeFromParentNode()
+        }
 
-        if let hit = hitResultsFeaturePoints.first {
-            let anchor = ARAnchor(transform: hit.worldTransform)
-            sceneView.session.add(anchor: anchor)
+        if let nodeExists = sceneView.scene.rootNode.childNode(withName: currentTextName, recursively: true) {
+            nodeExists.removeFromParentNode()
```

Comment:
**Minor naming tweak:**
In its current state I would probably call this function `removeChildNodes()`

I might use the single liners below, although I don't know 100% where Novoda stands on that :-)
```suggestion
private func removeChildNodes() {
    sceneView.scene.rootNode.childNode(withName: currentCubeName, recursively: true)?.removeFromParentNode()
    sceneView.scene.rootNode.childNode(withName: currentTextName, recursively: true)?.removeFromParentNode()
}
```

or perhaps more generically:
```suggestion
for childNode in sceneView.scene.rootNode.childNodes {
    childNode.removeFromParentNode()
}
```

\+ The rest of the code where there is `if let nodeExists`

Reply:
The issue is that I only want to remove if the cube we want to place is the same that already exists. This shows up to the user as moving that cube. But we have 3 possible ones we can add and "remove" so I can't do it for all of them, this code is to remove a model if we are trying to position the same one on the scene only. 

Which is why I named it `removeNodeIfExistAlready`

Reply:
Ahhh I see, I didn’t register how the currentNode/currentText was setup.

Maybe `removeCurrentNodes()` is what I mean. The other comments I made don’t matter in that case, thanks for clearing that up!

Leave it if you don’t feel it’s necessary :)